packages/workbox-build/src/entry-points/generate-sw.js

```diff
@@ -0,0 +1,74 @@
+const path = require('path');
+
+const GenerateSWOptions = require('./options/generate-sw-options');
+const copyWorkboxSW = require('../lib/copy-workbox-sw');
+const getFileManifestEntries = require('../lib/get-file-manifest-entries');
+const useBuildType = require('../lib/use-build-type');
+const writeServiceWorkerUsingDefaultTemplate =
+  require('../lib/write-sw-using-default-template');
+
+// TODO: Change to official CDN URL.
+const CDN_URL = `https://unpkg.com/workbox-sw@2.0.3/build/importScripts/` +
+  `workbox-sw.prod.v2.0.3.js`;
+
+/**
+ * This method creates a list of URLs to precache, referred to as a "precache
+ * manifest", based on the options you provide.
+ *
+ * It also takes in additional options that configures the service worker's
+ * behavior, like any `runtimeCaching` rules it should use.
+ *
+ * Based on the precache manifest and the additional configuration, it writes
+ * a ready-to-use service worker file to disk at `swDest`.
+ *
+ * @param {Object} input
+ * @return {Promise<{count: Number, size: Number}>} A promise that resolves once
+ * the service worker file has been written to `swDest`. The `size` property
+ * contains the aggregate size of all the precached entries, in bytes, and the
+ * `count` property contains the total number of precached entries.
+ *
+ * @memberof module:workbox-build
+ */
+async function generateSW(input) {
+  const options = new GenerateSWOptions(input);
+
+  const destDirectory = path.dirname(options.swDest);
+  const buildType = (process.env.NODE_ENV &&
```

Comment:
This logic will cause the `dev` build of WorkboxSW (either from the CDN or locally copying it) to be used iff `NODE_ENV` starts with `'dev'`. It was very difficult for folks who were using `generateSW()` in v2 to opt-in to using the `dev` build, so hopefully this will make it easier for folks to benefit from the extra debugging in their own dev server.

If folks are concerned about this approach, then I can make it so that developers explicitly pass in an option to accomplish the same thing instead of inferring it.